src/main/java/jenkins/plugins/jobcacher/ArbitraryFileCache.java

```diff
@@ -60,6 +63,7 @@ public class ArbitraryFileCache extends Cache {
     private static final long serialVersionUID = 1L;
 
     private static final String CACHE_VALIDITY_DECIDING_FILE_HASH_FILE_EXTENSION = ".hash";
+    private static final String CACHE_VALIDITY_DECIDING_FILE_DIGEST_ALGORITHM = "MD5";
```

Comment:
shouldn't `SHA256` be used to reduce chances of collisions?

Reply:
I was thinking about this too, but this will invalidate all caches, as Jenkins is using MD5 per default if you calculate the digest of a file. Maybe we could introduce a fallback mechanism to not break existing caches. What do you think?

Reply:
you're right, let's be compatible for now. 🤗

thanks for merging